feat: Rethrow original exception in kotlin invisible calls

Gabriel-Moraes350 opened the pull request:
- Add support for suspending function
- Add examples throwing exceptions

I am kind of divided between implementing it on Java as well, because of the checked exception.
What do you think does it makes more sense: To add almost the same behavior on Java, or to treat it differently, using a CatadioptreException that is an unchecked exception, and we tell the users to compare with the `cause` inside our custom exception?

Here in the CI, it is possible to see the problem in the Java examples ( it was the same behavior on Kotlin, before the change)

ericjesse commented:
Hello Gabriel, the idea is really good. I would also do the same in Java, of course. However, the checked exceptions might need to be wrapped anyway in an unchecked one anyway...

Gabriel-Moraes350 commented:
I've created a new issue there, just for tracking .
And updated the PR with a new unchecked exception created to hold the original exception. So I changed the previous Kotlin implementation to be more similar to Java in this aspect and when/if we change the code to be one for both, the behavior would remain the same

Gabriel-Moraes350 commented:
@ericjesse Can you please, review it again?
